ssl-config/build.gradle

```diff
@@ -21,3 +29,54 @@ task generateCerts(type:Exec) {
 
 tasks.check.dependsOn(javadoc)
 tasks.test.dependsOn(generateCerts)
+
+dockerTestRunner {
+    // missing BouncyCastle but contains Sun libraries
+    dockerRunner name: 'openjdk-7', dockerFile: file('src/test/resources/openjdk-7.dockerfile')
+
+    // missing BouncyCastle and Sun libraries
+    dockerRunner name: 'ibmjdk-8', dockerFile: file('src/test/resources/ibmjdk-8.dockerfile')
+
+    // run only the PKCS1 tests and exclude BouncyCastle runtime dependency
+    testConfig = {
+        String testsPattern = '**/pkcs1/*Tests*'
+        include(testsPattern)
+        classpath = classpath - getFilesForDependency('testRuntime', bcpkix)
+
+        onlyIf {
+            // verify that include pattern matches files
+            if (project.fileTree(getTestClassesDir().absolutePath).include(testsPattern).isEmpty()) {
+                throw new GradleException("No tests matching pattern '${testsPattern}' were found.")
+            }
+            return true
+        }
+    }
+
+    if (System.env.CIRCLE_TEST_REPORTS) {
+        runArgs.put('-v', "${System.env.CIRCLE_TEST_REPORTS}:/reports")
+        runArgs.put('-e', 'CIRCLE_TEST_REPORTS=/reports')
+    }
+}
+
+/**
+ * Returns a FileCollection that contains all of the files provided by the provided dependency in the given
+ * configuration.
+ */
+private FileCollection getFilesForDependency(String configName, String dependencyString) {
```

Comment:
probably worth thinking about just adding this to dockerTestRunner if it's going to be used in places where you expect to use dockerTestRunner